### PR TITLE
amp-fit-text: avoid innerHTML serialize/parse

### PR DESCRIPTION
**Pull request by samouri**
**summary**

Using the clone API is both faster and more worker-dom friendly than using `innerHTML`

cc @ampproject/wg-performance

### CODE REVIEW

File: extensions/amp-fit-text/0.1/amp-fit-text.js [hunk 1]
```diff
@@ -241,13 +242,29 @@ export function buildDom(document, element) {
...
+ * Make a destination node a clone of the source.
+ *
+ * @param {!Node} source
+ * @param {!Node} dest
+ */
+function mirrorNode(source, dest) {
+  // First clear out the destination node.
+  removeChildren(dest);
+
+  const {childNodes} = source;
```
Comment (samouri) [inline extensions/amp-fit-text/0.1/amp-fit-text.js:262]: nit: add a comment for:
`// Then copy all the source's child nodes  into destination node.`

**Review by rcebulko — APPROVED** [review]

**Review by dmanek — APPROVED** [review]

File: extensions/amp-fit-text/0.1/amp-fit-text.js [hunk 2]
```diff
@@ -241,13 +242,30 @@ export function buildDom(document, element) {
...
+ * @param {!Node} source
+ * @param {!Node} dest
+ */
+function mirrorNode(source, dest) {
+  // First clear out the destination node.
+  removeChildren(dest);
+
+  // Then copy all the source's child nodes into destination node.
+  const {childNodes} = source;
+  for (let i = 0; i < childNodes.length; i++) {
```
Comment (jridgewell) [inline extensions/amp-fit-text/0.1/amp-fit-text.js:264]: Nit: `childNodes.length` is a live-updating reference that exists in C++, which means you're calling from JS into C++ during every iteration (slow). Cache `length` locally, or use `nextSibling` iteration.

Comment (samouri) [reply extensions/amp-fit-text/0.1/amp-fit-text.js]: I've never done a `nextSibling` iteration before....so I cached length

**Comment by samouri** [comment]
For those curious, I created a benchmark. This method is [~25% faster](https://jsbench.me/g8kr12qtta/1).

Screen Shot 2021-07-12 at 4 26 07 PM

File: extensions/amp-fit-text/0.1/amp-fit-text.js [hunk 3]
```diff
@@ -241,13 +242,27 @@ export function buildDom(document, element) {
...
+ *
+ * @param {!Node} from
+ * @param {!Node} to
+ */
+function mirrorNode(from, to) {
+  // First clear out the destination node.
+  removeChildren(to);
+
+  // Then copy all the source's child nodes into destination node.
+  copyChildren(from, to);
```
Comment (rcebulko) [inline extensions/amp-fit-text/0.1/amp-fit-text.js:263]: This already existed? How convenient! :o

**Review by rcebulko — APPROVED** [review]

**Pull request merged**